### ISSUE TITLE
How To Add/Customize Titles and Labels

### ISSUE THREAD

**Issue by Jayboy75** [react-bootstrap-table/react-bootstrap-table2#873]
Filters and search bar don't have label tags, which is a requirement for accessibility. Pagination buttons have titles numbered "1" to "x" but aren't customizable, which results in title text being a duplicate of the element text - also not accessible. Is there a way in the current package to customize any of these elements by adding labels to form inputs or modifying title tags? If so, how could this be accomplished?

**Comment by AllenFang** [comment]
> Filters and search bar don't have label tags,

Can you explain it more and give it an example, I can implement it, thanks

**Comment by Jayboy75** [comment]
@AllenFang In order for a form control (in this case a text input or select dropdown) to be accessible, it must have an associated label tag which describes its purpose to users of assistive technology. You can hide the label visually using the Bootstrap "sr-only" (screen reader only) tag, which will make it available for those who need it without altering the UI.

Here's what I'm using for my custom search bar to make it 508-compliant:

```
<label htmlFor={`table-search-bar-${tableId}`} className="w-100">
        <span className="sr-only">Search This Data Table</span>
        <input
          id={`table-search-bar-${tableId}`}
          className="form-control form-control-sm"
          ref={n => input = n}
          type="text"
          placeholder="Search"
          onChange={handleChange}
        />
</label>
```

Notes on the above: I'm passing a 'tableId' prop because I have multiple tables per page, and each 'htmlFor' - 'id' pair must be unique. The text inside the span tag is the actual label describing the form control's purpose, hidden from view to normal users with the 'sr-only' bootstrap class.

A custom select component (column filter dropdown), if I could build it, would look something like this:

```
<label htmlFor={`filter-column-${column.title}`}>
          <span className="sr-only">{column.title}</span>
          <select id={`filter-column-${column.title}`} onChange={handleChange}>
            {column.options.map(val => (
              <option value={val.title}>{val.title}</option>
            ))}
          </select>
</label>
```

...with column.title, column.options, and val.title being pseudo-code for their respective properties.

Another area of react-bootstrap-table-2 that could use a simple fix would be in the pagination buttons, which render with a default title attribute equal to the page, ie., for the button with text "1" it renders title="1". Any title tags used should differ from the text content of the element. So for example if these were changed to title="page-1", title="page-2" etc. this would meet the requirements for accessibility (of course, the title attribute could be removed entirely as well, as the main requirement is that IF title tags are used they must not be identical or redundant to the element text or ALT text).

Let me know if I can clarify anything further. Thanks again for your attention to this!

**Comment by AllenFang** [comment]
@Jayboy75 thanks your explanation, that is clear for me and I enhance this when I have time, sorry I'm busy recently, so PR is also welcome 👍

**Comment by Jayboy75** [comment]
@AllenFang sure thing, pull request submitted - https://github.com/react-bootstrap-table/react-bootstrap-table2/pull/890

**Comment by AllenFang** [comment]
[RELEASE NOTE](https://react-bootstrap-table.github.io/react-bootstrap-table2/blog/2019/04/16/version-bump.html)

Please update related packages listed in above notes, thanks

**Comment by jmmadsen** [comment]
@AllenFang @Jayboy75 I recently upgrade my `react-bootstrap-table2-paginator` module to 2.0.6, but I am still getting this accessibility warning - `IDs used in ARIA and labels must be unique`. Do you know if your PR addresses this? Perhaps I upgraded incorrectly.

**Comment by jmmadsen** [comment]
I am getting this warning on the pagination size dropdown button, and I am just using the standard paginationFactory

**Comment by Jayboy75** [comment]
@jmmadsen My PR didn't alter the pagination buttons specifically - I just built out my own custom pagination buttons, which you can do using examples in the docs.

**Comment by jmmadsen** [comment]
That makes sense, thank you.
